### PR TITLE
Add automated linting

### PR DESCRIPTION
**Pull request by EllaBellaStella**
I want to deal with linting and other housekeeping type things with this first PR but I'm just gonna jot down some other stuff I noticed about the unreleased new calendar:

 - there's actually JS for that calendar, and it seems a bit overkill and as far as I can tell is to do the logos, which seems like a media-query thing to me
 - some responsiveness issues with the calendar, at about `lg` breakpoint and under it starts to jank, but nothing too severe, at least with the current content. I also don't know under what context it's gonna be deployed (what are the settings of its parent container, etc.)
   - day name header column
   - logos interrupt text instead of disappearing
   - the "Official Events" sidebar should have `min-width` to prevent excessive reflow
 - <del>wait why does this thing have scss?</del>
 - semantics
   - I find myself thinking that a naive user will find the sidebar confusing
     - the use of icons as the only label of what a piece of text is supposed to mean goes against web semantics and common practices
     - there's a lot of jargon that only an insider would understand e.g. "Special" and "Extra" mean nothing to the naive user, and only attains any meaning when they scan further down and see "Recurring Schedule"
     - TBH even as an insider I have no idea how "Extra" is different from "Special". Does the distinction warrant all this visual and cognitive noise?
     - centering the title is not the common practice and creates visual noise, the horizontal dividers do this as well
   - over-repetition, e.g. Monday Session is Monday 3rd/17th/24th, Tuesday Session is 
Tuesday 4th/11th/18th/25th, `[dollar]` icon followed by `$2.50`, and so on, and it happens in the calendar as well. Also calling a session by its day of the week is a jargon-y thing that a regular gets; the naive user would probably prefer the location to jump out at them instead
    - the jargon-y thing and the all-inning on icons interact in some funny ways from the naive users' perspective like: `[no money]` icon followed by `$2 donation requested`. This does *NOT* make sense to an outsider. In fact reconsider the phrase "donation _requested_" what does that even mean???
    - the `[recurring event]` icon in the calendar is redundant - that the events recur is self-evident - this robs almost all meaning from the other icons that appear in the calendar as well
    - calling these events recurring is a misnomer to the outsider because they defy the common understanding of recurrence - the naive user thinks of the options presented to them in the dropdown menu of their phone's calendar app; our events in contrast blatantly skip days in ways which again to the outsider seem to be for no reason at all. We should not force an insider's understanding of recurrence on ALL users
  - global style issues
     - overuse of `bold` font weight creates visual noise and steals valuable empty space
     - it is not clear (not even to me) why the colors for each event are not chosen uniquely; if color reuse was intentional, it's *very* unclear to me as to what they even mean
       - okay I just saw that the colors are unique but you are talking about someone who scored 200/200 in the Pantone color discrimination test and I only saw it on a monitor that costs as much as my first car, so I think the point still holds...
     - the over-under color choices could be a bit smoother (e.g. there is black on blue but there is also white on blue), using a design system generator will probably help here
     - the logos that appear in the calendar do not actually add any information but do add the cost that at every single breakpoint the logos intrude into the title of the event itself and only stop doing this at an impossible breakpoint of `2700px` and the sidebar is programmatically removed. At this point they actually look _really nice_. It also occurred to me that the colors chosen and a lot of the visual noise I complained about suddenly melt away at this impossibly high resolution (> 4.5K) and P3 color gamut. Unfortunately this is not what the majority of our users will be seeing

### CODE REVIEW

**Comment by Dislexsick** [comment]
Heya, thanks for the feedback! 😄 

I think it's worth explaining a bit more about how this is calendar is made, and some of the other decisions that go into it, which might explain some of the outcomes.   There are many things that can be changed, a few that cannot be, and at least a few I'm not sure on.

This is v0.1 of this - I hit a decision point earlier this week, it was either post this as-is for June, as we were already a few days into the month, or also go back and make the older version of it which is produced out of MS Word -> PDF exported -> Screenshot that PDF to give us images.
So far the images have been the only output of this, not even sharing the PDF (which tbeh was maybe a feasible option)

That word calendar is a janky mess, too -- it started as a macro'd template, but I very quickly broke it so now I need to do the numbers manually.   Here's a back-stage view of my little hell with this.  Word tables are not fun :'( 
image

We'd also had a request for the Southern events to be included in the calendar too, and a currently unannounced extra event planned -- This created a requirement that some days show two events, which just wasn't feasible with the current design (well, not with the amount of effort I was willing to give it).

So I set out to re-invent it; I wanted the creation of it to be automated, and it now needed to support multiple events on each day.  
When I sat down to start this I realised creating it as a webpage would let me continue to produce the images I wanted, but also allow it to be posted online in the future -- it's critical to understand this wasn't designed to be a webpage first, though I understand that there are plenty of items of feedback that apply to it regardless of how it's shown.

tl;dr: My task list is this atm 🤣
[x] Create a calendar that shows 100% of content in a single view so it can be used as an image.
[ ] Make it work as a good webpage 

This is roughly how it's supposed to look as an image -- ofc it's not final, but it's like 80-90% of the way there
image

Notable: that image is >2000px wide!  That will become relevant later ;)

As soon as I started the process the first thing I started with was getting my data in, so I could have some events to work with -- but I qauickly realised getting the events as an input from a file (e.g. loading from CSV/XML/JSON etc) was going to be painful as it meant coding systems for definign recurring events, but cancelled instances, and whatnot. 
Solution: Use Outlook to handle most of it.  Honestly, if I could change the view a little bit a screenshot of the Outlook calendar almost works to fit many of my goals...  Alas, I wanted more key details visible to those looking at the calendar.

image

So I very quickly worked through that first challenge, and the event data is incoming via MS Graph; and started working on getting it into a calendar.  

A design goal of this was to have it divided up into a calendar that contains the very core details about an event; then an agenda on the side which provides the rest of the information that an attendee needs to make a decision about whether they want to, and/or attend an event (e.g. Cost, location, timing, etc).

## A ~~quick~~ break for a sidebar about terminology I've used within this
There are terms and concepts used in the calendar that the committee have barely discussed, let alone us introduce to the community, or push to the point of being well-known 

- ABGG official vs. Community.  That's already explained [here](https://abgg.au/#events), but I would admit it's almost defs not well-known.
- Event types of Special / Extra / Recurring Schedule -- This is a brand new concept, and honestly those three terms are even newer.  This was their debut, but they very well may not be the final terms... they're just what I had down when I made the call that in the battle of "it's good enough" vs. "we're HOW far into the month already?!?" that it was time to pull the trigger on posting it

ABGG has previously unofficially had kinda two tiers of events

1. Special Events
3. Regular events

Under the two tier system anything that wasn't a regular event is a special event.

Recently we've moved to essentially three tiers, though what I'm discussing here I'm trying to lead, and doing so by example -- not even our whole committee are on the same page with this 😅

In the three tier system we have
1. Two tentpole events planned each year - International Tabletop Day in June, and our end of year event in December.  These events are special events taken up a notch, where our entire committee are on deck organising it, and it takes real effort to make happen.  As such we want to make them the stars of the show - if you can only make two ABGG events a year then these are the ones to make!  These have taken over the moniker of special events, though this is not consistent (e.g. If you look at the FB cover image for the King's Birthday event it's labelled special event.
  - Following on from the notion of "...if you can only make two ABGG events a year..." this also means if there's one event on the calendar we want people to remember it's these.
  - I didn't do a good job of this in v0.1 -- with the next tier intruding on the visual space this event was intended to occupy
3. Events that break the schedule in some sort of way, or are rare (once a year), but don't require a lot of extra effort (e.g. Public Holiday sessions).  Previously these were "special" events, but some feedback from the community was "What makes them special?" especially as over time the committee can't always put the same effort into them.  These have been temporarily dubbed "Extra" events.  Notably ABGG typically only run one official event on any given day, so an extra event will also typically replace teh 
4. Events that occur on a regular, predictable schedule at least ~once a month.
 
## Behind the scenes in the script that generates this

There are MANY things going on behind the scenes in the generation of this - for instance 

- There are a bumch of small tweaks to handle things that otherwise look weird, but then make things inconsistent.  e.g. If on the calendar the text would read "At X by X" it instead just uses "At X" ("By The Lost Dice At The Lost Dice" vs. "At The Lost Dice") -- right now this means all events for TLD are slightly different.
- It's designed that the Official ABGG events will always sort before community events; then within each group the events will be sorted Special > Extra > Regular... on that note...
- You may have noticed empty divs with the class agenda-event-grid-1col -- what's up with them?  ...and why are they sometimes not empty?  Well, that's a quick and dirty hack -- Content is (mostly) divided up based on the number of characters in them -- content that's too long gets put into the second div with "1 column" (Two, really - icon + content -- but it's labelled based on the content... because that's what I wrote at the time).  Content is placed in an order relative to each other, and there's functionality to force some content into the 2col section even if too long (e.g. Event name, and who's running it, so they always appear first)

image


## Javascript
- The JS is there to size the logos based on the smallest child within the calendar, and agenda respectively.  This prevents two items with the same logo having logos of differing sizes, which looks weird.  I couldn't work out any way to do this without JS, and honestly that JS came together quickly and easily.  
  - Specifically the JS goes through all elements with the classes '.agenda-event.logo-image' (and later repeats for '.calendar-event.logo-image' .. with duplicated code, but idgaf enough to functionise that lol); it seeks out the smallest; then sets the CSS var --height on all of them to be that same value.  The whole usage of a CSS var is to work around being able to access and change styles of psuedo-elements easily.
  - Now's probably a good time to mention for some of the decisions I've made while making this I'm going to point the finger and say either "[AI|StackExchange] told me to do it that way" :P 

## Webpage layout, semantics
- There is a lot of work needed for how the whole page gets sized when rendered as a webpage -- right now most of the settings have been designed to work on my screen (2560p, which is also a problem since it results in images that are quite large too).  
  - I've already gone through and fixed the headers, this was one of the first things I done and I had less experience with the grid model... the reason the headers tear off is because it's really two grids, and most of the time they stay in sync just fine.
  - Before I published this I actually went through and ripped out a lot of the stuff I had in CSS that was contraining widths, and added a lot more flex in things, without really knowing the impact for all of them... that's likely resulting in a lot of jank now.
- Icons on the sidebar are damn near required for this to work as a single static image, though I very much agree otherwise --this is not accessible and would need to be addressed for this to be a page -- but as soon as it moves to a page we gain options anyway.

## Headers for Special/Extra/Recurring
- Honestly the headers that showed Special/Extra/Recurring were one of the things I'd done the least with so far, and I wasn't happy with, but wasn't quite sure where to go with them. 
  - I've turned them off for now, I'll mull over what I want to do with them.  As for jargon, re "only attains any meaning when they scan further down" -- would that be adquately handled if it were something like "Special events", "Extra events" (or something to that effect)?
  - I'm not totally happy with them off, as I like the utility of people being able to quickly identify what is not a routine event, but we can figure something out for them.

## Repetition of things
-  Repetition on things like the dates and times is intentional, and I have no real intention on changing that -- the agenda sidebar is supposed to be an exhaustive list, with quick reference of all key details.
  - Re-reading though, is  one of the issues there that the event is called "DayOfWeek Session" then we soon after list "DayOfWeek <ListOfDates>" thus repeating the DayOfWeek text in close proximity?  If so I have to think about how we handle this.  This will only be an issue for ABGG official events, as all others have more distinct names.  If I named them after the venue we'd also repeat that in close proximity in the calendar -- and also I did previously have them named that way and IIRC someone asked me to change it🙃
  - tbeh, special handling for ABGG events is possible, and for the regular events we may be able to just have "ABGG at <Venue>" on the top line.  I do feel that naming them "DayOfWeek Session", or not is high on priorities to solve though.  Right now it reinforces to people just looking at it for the first time that it's repeating, and it's one of several, though if they spend a few more seconds this would indeed be an outcome they should arrive at regardless. 

## Dollarydoos
  - I very much agree with the dollar icon followed by dollar symbol.  I couldn't make that look nice either way, so went with duplication for v0.1 and it became a future me problem to work out what to do there.
  - The `[no money]` icon will automatically get used when the cost contains the substring "Free" or "donation" -- In the case of the latter a donation will always be _requested_, but never _required_.  The event, for those that don't donat is free thus the usage of the icon to represent this.  
  - The phrasing on "donation _requested_" is unlikely to change, it's an interaction with how the council charges for use of their facilties -- as soon as we charge people we start paying a prohibitively expensive hire rate; but we also have public liability insurance we need to pay for -- so want to collect money if possible.  The council allows us to request people donate, and they're satisfied with this phrasing as indicating it's not mandatory.  

## Recurrance stuff
- I've just made a version of the calendar that drops the icon, if it would be the recurring icon, and that buys back a lot of space and makes the other icons significantly more impactful.  
- I'm not sure I can agree with the last bulllet under semantics... or I'm not following what you're saying.
  - One takeaway I have from it, whether it's what you intended or not, was there's a ferature that I intended on adding, but forgot about until now -- a human-readable recurrance pattern.  e.g. Real-world examples of three patterns we have, that I want to represnet this way are "Every Tuesday"; "Fortnightly, on Wednesdays" "First Saturday of each month"
  - I think someone thinking about the options of the dropdown in their phone calendar is good, that's kinda what we want?
- I will, when I get a chance, PoC having a much smaller recurring icon against the relevant calendar entries, because it's a design language that's used commonly elsewhere, and I very specifically want to evoke that when people are looking at this (which is why I'm also confused about the phone drop-down part - I want people making the comparison)
image
- Regarding skipping instances for no reason, is that central to the point?  My read of what you're getting at there is tthat someone looking at this for the first time sees an event marked as recurring, but then doesn't see it in one of the slots this month they'd expect it to? (...nor in the slots that are shown for the non-current months... another problem on the to-do list I need to handle).  e.g. Parks is recurring, but doesn't appear on three of the six Monday slots that are visible?
    - I already have two things kicking around in my head for this -- One is cancelled event handling, whether I want to specifically show the absence of an instance that would usually occur we've had mixed feedback about this, but I want to PoC some styling I think will work -- a big ol' red X with some strikethrough on the event name ought to help!  (...right?!?).
    - The other item is rendering in events on the "inactive month" cells of the calendar, but this requires several changes; and design decisions (e.g. Do they get styled differently, such as washed out to show they're a different month?; whether to show [and how, if so], or hide these from the agenda?)

## Styles

- Bold: No styles are final...  but I didn't think there was too much bold 🥺 -- anywhere in particular you think it can be culled?

### Colours

Oh lawdy, they're fun /s 😅😅   

- Most are still just whatever I applied at the time, there are many tweaks needed, though it's worth noting the selections made so far are not random for the most part....
  - Up until a few hours before I posted it all community events were visually distinguished by the presence of a border, where ABGG events were not.  I ended up thinking that looked weird, so I decided the ABGG events needed a border too, and so quickly just applied one which uses CSS color-mix to darken the base colour.
- The color schemes for each event are based on the "branding" colours for either the people that manage the event, or the venue.
  - In the case of ABGG I have a system of colours being associated with venues, so that is used -- except then we're left with an event that's not at a gaming venue, so it uses the ABGG logo blue... 
  - Many of the community events pull colours from their brand, e.g. AVCon uses colours from their site for this years event; and GAMES uses the colours found naturally in their logo
  - Others get a bit weirder -- South Side Games doesn't really have a colour scheme as they're still new... Jordan whipped up their logo for me on-request, and that was after I'd set the colour for their event based on venue, and I used the orange from San Churro's brand 😆
- Usage of black text vs. white text, even when looking at the same group, same venue, comes from a desire to do things to visually distinguish the special and extra events -- there's a LOT of scope to improve things there -- this was one of those things that made it through thanks to the battle of time spent so far vs. needing to deliver *_something_* to the group for June.
- Seeting expectations: You're talking with someone who's colourblind, and previously had a(n uncalibrated) monitor that should have been relatively colour accurate... but that died, sooooo now I'm just using what I'm using 😅😅😅

Despite the above note on colourblindness, there's something of a hot take for colours, and almost the design of this in general: We're a volunteer-run community group, we're allowed to be scrappy on this - in fact it's in our best interest to NOT go too professional looking on this, else we come across to people as something we're not.  This also needs to be maintainable by future generations that run the group, without them also needing someone skilled in design.

I'm happy for tweaks (and there's scope for plenty of them) but we wont be able to stray away from some core constraints -- such as the way the core colours are picked-- which I expect could cause creating a harmonious design damn near impossible, if not at least quite hard.

image
image

### Logos

I really like these, and they've been so much effort I'm retiscent to ever remove them :P 

I mentioned way earlier that the image I posted is >2k pixels wide, and it would be relevant later... 
That's because outside of me finding one spot where the titles of events in the calendar broke onto two lines for many isntances, and everything else worked welll (which is what I then took the screenshot of) the calendar so far has been designed workign with close to that 2700px (2560p - some space for browser UI) breakpoint you mentioned -- that's approimately the only two sizes that this **_really_** works at right now, and yeah, I very much realise that's a problem that will need to be solved before this could be published as a page :)

**Comment by EllaBellaStella** [comment]
@Dislexsick Holy crap I just saw your comment, I'ma actually have to find time to read through it, and I'll let you know when I have. I didn't expect a comment on a draft PR (much less an essay of one)

Why is this PR here? So, the only way for me to see a somewhat decent diff _is_ to fork it and put up a PR back to this branch, technically that is the sole purpose (one could say I could do the same over on my fork but there is no good reason to do the extra work of keeping 2 PRs around).

Usually when my PRs are in draft I don't expect anyone to look at it, too seriously or even at all, the reason being a draft PR can change or disappear at any time. That goes for everything on top of changes - title, description, you name it.

So the notes in the description really _were_ just for me - I was going to further refine/form them and then file proper [issues](https://github.com/Dislexsick/ABGG/issues) - because frankly that is the only sane way to discuss, manage, and triage updates to an open source project that is owned by an org with as many potential stakeholders as ABGG.

So again...let me get back to you after I have had the time to take a proper lookin at what's above.

**Comment by Dislexsick** [comment]
Okay, fair, though it's worth pointing out at this point it is just you and I here atm... I'd put the pages on GitHub pages mostly because it's easy free hosting, but at least a little to collaborate... I'd just never layed the groundwork for the latter 😅

Hopefully the above essay is useful when you have time to review it though :)

**Comment by EllaBellaStella** [comment]
@Dislexsick Github is _the_ choice since git on local only loses most of its point in your case, I would say.

Regardless of Github, it's the choice of public vs. private repo which makes it open source (supported by what's in the README), and it's not an unchangeable choice.

What is usually considered immutable is the git (commit) history, which unlike Github comments and the like which are ephemeral, literally everyone who will ever work on this project is "with us on it"; I tend to assume it includes everyone on this planet with an internet connection for the duration it is pubic/open sourced.

For these reasons I tend towards a "never too early" approach over getting a project organized and I have found the readability pays off for future contributors (future me included) sooner than one would think.

**Comment by EllaBellaStella** [comment]
Case in point conversations like these would live in an issue instead (if one did exist) and there because the tool fits the purpose they are _not_ treated ephemerally by all projects, esp open ones
